File: server/src/main/java/com/nhannn/generic_ecom/models/apis/sign_up/SignUpRequest.java
package com.nhannn.generic_ecom.models.apis.sign_up;

import com.nhannn.generic_ecom.models.User;
import com.nhannn.generic_ecom.models.apis.BaseRequest;

/**
 * Author: nhannn
 */
public class SignUpRequest extends BaseRequest {
    User user;

    public SignUpRequest() {
    }

    public SignUpRequest(User user) {
        this.user = user;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: ui/src/app/models/api/sign-up/sign-up.request.ts
import { BaseRequest } from "../base.request";
import { User } from "../../user.model";

/**
 * Author: nhannn
 */
export class SignUpRequest extends BaseRequest {
  user: User;

  constructor(user: User) {
    super();
    this.user = user;
  }
}
